File: Additional/1232.Check If It Is a Straight Line.py
class Solution:
    def checkStraightLine(self, a: List[List[int]]) -> bool:
        for i in range(1,len(a)-1):
            if a[i][0]-a[0][0] == 0 and a[i+1][0]-a[0][0] != 0:
                return False
            elif a[i][0]-a[0][0] != 0 and a[i+1][0]-a[0][0] != 0:
                if (a[i][1]-a[0][1])/(a[i][0]-a[0][0]) != (a[i+1][1]-a[0][1])/(a[i+1][0]-a[0][0]):
                    return False
        return True

File: Additional/1232.Check If It Is a Straight Line.ts
function checkStraightLine(coordinates: number[][]): boolean {
  const [x1, y1] = coordinates[0];
  const [x2, y2] = coordinates[1];
  const standardSlope = x2 > x1 ? (x2 - x1) / (y2 - y1) : (x1 - x2) / (y1 - y2);
  for (let idx = 2; idx <= coordinates.length - 1; idx++) {
    const [x1, y1] = coordinates[idx - 1];
    const [x2, y2] = coordinates[idx];
    const curSlope = x2 > x1 ? (x2 - x1) / (y2 - y1) : (x1 - x2) / (y1 - y2);
    if (curSlope != standardSlope) return false;
  }
  return true;
};
